scripts/create_dataset.py

```diff
@@ -1,69 +1,160 @@
+    :param column_array_of: indicates if the column values should be in array
+    :param input_table_name: the name of input JSON table that has the input_column
+    :param input_tables: the input tables that have been added to a dict and used to validate the created relationships
+    """
+    if not any([validate_table_and_column(table_name=input_relationship["table_name"],
+                                          column_name=input_relationship["column_name"],
+                                          column_datatype=column_datatype,
+                                          input_tables=input_tables)
+                for input_relationship in input_relationships] if input_relationships else [True]):
+        raise SystemExit("Error: The relationship(s) provided for table, {}, column {} are not valid"
+                         .format(input_table_name, column_name))
+
+    relationships = [
+        {
+            "name": "{}_{}_to_{}_{}".format(input_table_name,
+                                            column_name,
+                                            input_relationship["table_name"],
+                                            input_relationship["column_name"]),
+            "from": {"table": input_table_name, "column": column_name, "cardinality": "one"},
+            "to": {
+                    "table": input_relationship["table_name"],
+                    "column": input_relationship["column_name"],
+                    "cardinality": "many" if column_array_of else "one"
+                  }
+        }
+        for input_relationship in input_relationships] if input_relationships else []
+
+    column = {
+        "name": column_name,
+        "datatype": column_datatype,
+        "array_of": column_array_of is True
+    }
+
+    return column, relationships
+
+
+# TODO add validation for duplicate assets, tables, columns and relationships
+def validate_table_and_column(table_name, column_name, column_datatype, input_tables):
+    """
+    Validates a relationship checking the the given table and column exist in input tables and the datatype is correct.
+    :param table_name: the name of table containing has the input_column that will be validated against the input tables
+    :param column_name: the name of the input_column that will be validated against the input tables
+    :param column_datatype: the primitive type enforced for values of the input JSON column
+    :param input_tables: the input tables that have been added to a dict and used to validate the created relationships
+    """
+    for input_table_name in input_tables:
+        if input_table_name == table_name:
+            for input_column in input_tables[input_table_name]["columns"]:
+                if input_column["name"] == column_name:
+                    if input_column["datatype"] != column_datatype:
+                        return False
+                    return True
```

Comment:
Since this always returns, could simplify the `if` with two `return`s to:
```python
return input_column["datatype"] == column_datatype
```

Reply:
lol